gateway-service-impl/src/main/java/org/hypertrace/gateway/service/EntityTypesProvider.java

```diff
@@ -0,0 +1,55 @@
+package org.hypertrace.gateway.service;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.entity.type.service.client.EntityTypeServiceClient;
+import org.hypertrace.entity.type.service.v1.EntityType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EntityTypesProvider {
+  private static final Logger LOG = LoggerFactory.getLogger(EntityTypesProvider.class);
+  private static final int DEFAULT_CACHE_SIZE = 1024;
+  private static final int DEFAULT_EXPIRE_DURATION_MIN = 30; // 30 minutes
+
+  // TenantId to Entity types cache
+  private final LoadingCache<String, List<String>> tenantIdToEntityTypesCache;
```

Comment:
Can we use request context as cache key, instead of tenant?

Reply:
What's will be the advantage of that? 🤔